### ISSUE TITLE
`source-repository-package` does not seem to support a relative `location`

### ISSUE THREAD

**Issue by FedericoStra** [haskell/cabal#8925]
**Describe the bug**

I'm trying to depend on another local package by putting a [`source-repository-package` stanza](https://cabal.readthedocs.io/en/3.4/cabal-project.html#specifying-packages-from-remote-version-control-locations) in the `cabal.project` file. The dependency is in a sibling folder, so the natural way would be to specify it with a relative path:

```cabal
packages: .

source-repository-package
    type: git
    location: ../dependency
    tag: master
```

However, this fails with a blatantly false statement:

```
$ cabal build
fatal: repository './../dependency' does not exist
```

I've also tried replacing the location with all the following alternatives

```cabal
    location: ./../dependency
    location: file://../dependency
    location: file://./../dependency
```
but to no avail.

The only way I could make it work is by specifying an absolute path, with either of the following

```cabal
    location: /home/<username>/code/haskell/dependency
    location: file:///home/<username>/code/haskell/dependency
```

**Expected behavior**

I would expect relative paths to work in the `location` field of the `source-repository-package` stanza.

**System information**

```
$ uname -srio &&  cabal --version
Linux 6.2.10-200.fc37.x86_64 x86_64 GNU/Linux
cabal-install version 3.6.2.0
compiled using version 3.6.2.0 of the Cabal library
```

**Comment by fgaz** [comment]
As a workaround, if you don't mind cabal picking up the working tree including untracked files, you can add the relative path to `packages`

**Comment by FedericoStra** [comment]
> As a workaround, if you don't mind cabal picking up the working tree including untracked files, you can add the relative path to `packages`

Yes, yes, I'm aware of that alternative, but it seems quite different to me:

- using `packages` uses the most up-to-date version as it happens to be in the current checked out state;
- using `source-repository-package` freezes the dependency to its state at a particular point in time (if I specify a `tag` which is a tag or a commit).

I encountered this issue because I needed to depend on a local package but I don't want the development of the dependency to break the dependent project.
